src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -103,54 +95,29 @@ public class RecordService {
   // This class is designed to work with empty metadata config map
   private SnowflakeMetadataConfig metadataConfig = new SnowflakeMetadataConfig();
 
-  RecordService(Clock clock) {
+  RecordService(
+      Clock clock,
+      boolean enableSchematization,
+      StreamingRecordMapper streamingRecordMapper,
+      ObjectMapper mapper) {
     this.clock = clock;
+    this.enableSchematization = enableSchematization;
```

Comment:
I wonder if `RecordService` needs to be aware of `enableSchematization` value after creating `StreamingRecordMapper` abstraction. In fact we could pass this flag to a mapper in a factory instead of `processSnowflakeRecord` argument.

Reply:
good idea